tests/integration-tests/tests/iam/test_iam.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import boto3
+import pytest
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+
+from tests.common.assertions import assert_no_errors_in_logs
+
+
+@pytest.mark.usefixtures("os", "instance")
+def test_iam_roles(
+    region, scheduler, common_pcluster_policies, role_factory, pcluster_config_reader, clusters_factory, cluster_model
+):
+    if scheduler == "awsbatch":
+        instance_policies = common_pcluster_policies["awsbatch_instance_policy"]
+        lambda_policies = common_pcluster_policies["awsbatch_lambda_policy"]
+    else:
+        instance_policies = common_pcluster_policies["traditional_instance_policy"]
+        lambda_policies = common_pcluster_policies["traditional_lambda_policy"]
+    cluster_role_name = role_factory("ec2", [instance_policies])
+    lambda_role_name = role_factory("lambda", [lambda_policies])
+    cluster_config = pcluster_config_reader(
+        config_file=cluster_model + ".ini", ec2_iam_role=cluster_role_name, iam_lambda_role=lambda_role_name
+    )
+    cluster = clusters_factory(cluster_config)
+    main_stack_name = "parallelcluster-" + cluster.name
+    cfn_client = boto3.client("cloudformation", region_name=region)
+    lambda_client = boto3.client("lambda", region_name=region)
+    # If scheduler is awsbatch, there will still be IAM roles created.
+    check_no_role_is_created = scheduler != "awsbatch"
+    _check_lambda_role(cfn_client, lambda_client, main_stack_name, lambda_role_name, check_no_role_is_created)
+
+    # Test updating the iam_lambda_role
+    updated_lambda_role_name = role_factory("lambda", [lambda_policies])
+    assert_that(updated_lambda_role_name).is_not_equal_to(lambda_role_name)
+    cluster.config_file = pcluster_config_reader(
+        config_file=cluster_model + ".update.ini",
+        ec2_iam_role=cluster_role_name,
+        iam_lambda_role=updated_lambda_role_name,
+    )
+    cluster.update(force=True)
```

Comment:
question: why do we need `force=True`?

Reply:
I imitated the code in `test_update`. In `test_update`, reason is that we don't stop compute nodes to update some parameters that require compute nodes stopped.
So on a second look, I don't have to use `force` here. I am testing the code without `force`, and will do the change if there is no failures